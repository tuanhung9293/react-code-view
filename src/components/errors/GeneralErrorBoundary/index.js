import React from 'react'
import InternalError from 'components/errors/InternalError'

class GeneralErrorBoundary extends React.Component {
  state = {
    hasError: false
  }

  componentDidCatch() {
    this.setState({hasError: true})
  }

  render() {
    if (this.state.hasError) {
      return <InternalError />
    }
    return this.props.children
  }
}

export default GeneralErrorBoundary