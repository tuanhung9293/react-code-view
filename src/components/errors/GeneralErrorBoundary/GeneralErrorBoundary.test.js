import React from 'react'
import {shallow} from 'enzyme'
import GeneralErrorBoundary from './'

describe('renders', () => {
  it('renders GeneralErrorBoundary in case without error', () => {
    expect(shallow(
        <GeneralErrorBoundary>
          I am a Child
        </GeneralErrorBoundary>
    )).toMatchSnapshot()
  })
})

