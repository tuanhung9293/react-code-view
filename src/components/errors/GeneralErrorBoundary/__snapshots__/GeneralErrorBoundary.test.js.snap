// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders GeneralErrorBoundary in case without error 1`] = `"I am a Child"`;
